src/common/Page.ts

```diff
@@ -2582,11 +2584,19 @@ export class Page extends EventEmitter {
   async evaluateOnNewDocument(
     pageFunction: Function | string,
     ...args: unknown[]
-  ): Promise<void> {
+  ): Promise<() => Promise<void>> {
     const source = helper.evaluationString(pageFunction, ...args);
-    await this._client.send('Page.addScriptToEvaluateOnNewDocument', {
-      source,
-    });
+    const { identifier } = await this._client.send(
```

Comment:
let's return the identified in evaluateOnNewDocument and create a symmetrical function `stopEvaluateOnNewDocument` (perhaps you have a better name?) This would make it more composable.

We would also need some tests that both functions work.